### PR TITLE
When Library add reference to NETStandardLib if not present

### PR DESCRIPTION
**Pull request by krwq**
Fix https://github.com/dotnet/cli/issues/4547

When NetStandardLibrary ref not present explicitly in project.json (i.e. present through transient dep) the generated msbuild was causing build errors

cc: @livarcocc

### CODE REVIEW

**Comment by mellinoe** [comment]
I would expect that a much smaller set of dependencies is actually needed to fix https://github.com/dotnet/cli/issues/4547. Probably just `System.Resources.ResourceManager` or something like that. Given the direction we're headed with the "metapackage", though, it might not matter that we're pulling in the whole thing.

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs [hunk 1]
```diff
@@ -13,18 +13,27 @@ internal static class ProjectExtensions
     {
         public static ProjectType GetProjectType(this Project project)
         {
-            var projectType = ProjectType.Console;
             if (project.IsWebProject())
             {
-                projectType = ProjectType.Web;
+                return ProjectType.Web;
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs]: I don't like this multiple return invocation pattern. Can you please revert this to the way it was before. With the assignment of the variable and a single return?

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs [hunk 2]
```diff
@@ -13,18 +13,27 @@ internal static class ProjectExtensions
...
+                return ProjectType.Library;
             }
 
-            return projectType;
+            return ProjectType.Console;
+        }
+
+        private static bool IsLibrary(this Project project)
+        {
+            return !project.IsTestProject && !project.GetCompilerOptions(null, "Debug").EmitEntryPoint.GetValueOrDefault();
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs]: We are now duplicating this code to generate the compiler options. Can we do that in the top method and pass it down?

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs [hunk 3]
```diff
@@ -169,8 +169,11 @@ private void CleanExistingPackageReferences(ProjectRootElement outputMSBuildProj
 
             AutoInjectImplicitProjectJsonAssemblyReferences(framework, packageDependencies);
 
+            bool hasNetStandardReference = false;
+            string netStandardRefName = "NETStandard.Library";
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs]: Can you put this in the PackagesConstant class?

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 4]
```diff
@@ -422,6 +422,18 @@ public void It_migrates_project_with_output_name()
                 .Count().Should().Be(1);
         }
 
+        [Fact]
+        public void It_migrates_and_builds_library_without_netstandard_reference()
+        {
+            string projectName = "LibraryWithoutNetStandardLibRef";
+            var projectDirectory = TestAssetsManager.CreateTestInstance(projectName, callingMethod: $"It_migrates_{projectName}").Path;
```
Comment (piotrpMSFT) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Why do we need to override the calling method? It's easier to see what's going on if it's just... the calling method

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 5]
```diff
@@ -422,6 +422,18 @@ public void It_migrates_project_with_output_name()
...
+
+            CleanBinObj(projectDirectory);
```
Comment (piotrpMSFT) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Why is this necessary? TAM will clean for you.

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs [hunk 6]
```diff
@@ -203,6 +206,21 @@ private void CleanExistingPackageReferences(ProjectRootElement outputMSBuildProj
                     itemGroup,
                     mergeExisting: true);
             }
+
+            if (project.GetProjectType() == ProjectType.Library && !hasNetStandardReference)
+            {
+                // int is meaningless since the template argument is not being used
+                AddItemTransform<int> transform = new AddItemTransform<int>(
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs]: Can you just use a PackageDependencyInfoTransform instead of creating your own? You just create a PackageInfo (which has three properties, out of which you need to populate two) and call PackageDependencyInfoTransform.

Much cleaner code than this.

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs [hunk 7]
```diff
@@ -13,59 +13,63 @@ internal static class ProjectExtensions
...
+                    projectType = ProjectType.Web;
+                }
+                else
+                {
+                    projectType = ProjectType.Console;
+                }
+            }
+            else
+            {
+                projectType = ProjectType.Library;
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs]: When creating the projectType variable above, you can just assign ProjectType.Library to it and then you can remove this else clause.

**Comment by piotrpMSFT** [comment]
:shipit:
